modules/core/src/main/java/org/apache/ignite/internal/processors/metric/export/VarIntWriter.java

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.metric.export;
+
+/**
+ * <p>Supports expandable byte buffer with varint put operations for {@code int} and {@code long} types.</p>
+ * <p>Each put operation moves position forward accordingly to the written type size.</p>
+ */
+public class VarIntWriter {
+    /** Backing byte array. */
+    private byte[] arr;
+
+    /** Current position. */
+    private int pos;
+
+    /**
+     * Constructor.
+     *
+     * @param initCap Initial backing byte array capacity.
+     */
+    public VarIntWriter(int initCap) {
+        arr = new byte[initCap];
+    }
+
+    /**
+     * Returns current position in backing byte array.
+     */
+    public int position() {
+        return pos;
+    }
+
+    /**
+     * Resets current position in backing byte array to {@code 0}.
+     */
+    public void reset() {
+        pos = 0;
+    }
+
+    /**
+     * Puts boolean value.
+     *
+     * @param val Value.
+     */
+    public void putBoolean(boolean val) {
+        ensureCapacity(1);
+
+        arr[pos++] = (byte)(val ? 1 : 0);
+    }
+
+    /**
+     * Puts double value.
+     *
+     * @param val Value.
+     */
+    public void putDouble(double val) {
+        ensureCapacity(Double.BYTES);
+
+        long v = Double.doubleToLongBits(val);
+
+        arr[pos++] = (byte)(v >> 56);
+        arr[pos++] = (byte)(v >> 48);
+        arr[pos++] = (byte)(v >> 40);
+        arr[pos++] = (byte)(v >> 32);
+        arr[pos++] = (byte)(v >> 24);
+        arr[pos++] = (byte)(v >> 16);
+        arr[pos++] = (byte)(v >> 8);
+        arr[pos++] = (byte)(v);
+    }
+
+    /**
+     * Puts integer value in varint format. It can consume from 1 to 5 bytes.
+     *
```

Comment:
this method is a very tricky, I think it make sense to provide more info about its implementation in javadocs.
May be a link to some internet resource about var int?
Like: https://learnmeabitcoin.com/guide/varint

Reply:
Added a short description to the class header.